controller/search_with_appending_results.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const _ = require('lodash');
+
+const searchService = require('../service/search');
+const logger = require('pelias-logger').get('api');
+const logging = require( '../helper/logging' );
+const retry = require('retry');
+const Debug = require('../helper/debug');
+
+function isRequestTimeout(err) {
+  return _.get(err, 'status') === 408;
+}
+
+function setup( apiConfig, esclient, query, should_execute ){
+  function controller( req, res, next ){
+    if (!should_execute(req, res)) {
+      return next();
+    }
+
+    const debugLog = new Debug('controller:search');
+
+    let cleanOutput = _.cloneDeep(req.clean);
+    if (logging.isDNT(req)) {
+      cleanOutput = logging.removeFields(cleanOutput);
+    }
+    // log clean parameters for stats
+    logger.info('[req]', 'endpoint=' + req.path, cleanOutput);
+
+    const renderedQuery = query(req.clean);
+
+    // if there's no query to call ES with, skip the service
+    if (_.isUndefined(renderedQuery)) {
+      debugLog.push(req, 'No query to call ES with. Skipping');
+      return next();
+    }
+
+    // options for retry
+    // maxRetries is from the API config with default of 3
+    // factor of 1 means that each retry attempt will esclient requestTimeout
+    const operationOptions = {
+      retries: _.get(apiConfig, 'requestRetries', 3),
+      factor: 1,
+      minTimeout: _.get(esclient, 'transport.requestTimeout')
+    };
+
+    // setup a new operation
+    const operation = retry.operation(operationOptions);
+
+    // elasticsearch command
+    const cmd = {
+      index: apiConfig.indexName,
+      searchType: 'dfs_query_then_fetch',
+      body: renderedQuery.body
+    };
+
+    logger.debug( '[ES req]', cmd );
+    debugLog.push(req, {ES_req: cmd});
+
+    operation.attempt((currentAttempt) => {
+      const initialTime = debugLog.beginTimer(req, `Attempt ${currentAttempt}`);
+      // query elasticsearch
+      searchService( esclient, cmd, function( err, docs, meta ){
+        // returns true if the operation should be attempted again
+        // (handles bookkeeping of maxRetries)
+        // only consider for status 408 (request timeout)
+        if (isRequestTimeout(err) && operation.retry(err)) {
+          logger.info(`request timed out on attempt ${currentAttempt}, retrying`);
+          debugLog.stopTimer(req, initialTime, 'request timed out, retrying');
+          return;
+        }
+
+        // if execution has gotten this far then one of three things happened:
+        // - the request didn't time out
```

Comment:
Should this read 'the request timed out'?

Reply:
It's correct as-is, the code block above this comment handles timeouts, so the code after handles "no timeout" and "there were too many timeouts so we've given up trying".

Reply:
Ah ok. so maybe the comment in that section on line 64 should read something like `return immediately if no further action should be taken because of a timeout or the retry limit being hit`, and then this comment can say something like `handl all errors (besides timeouts and retry limit hits which are both handled above)`. Does that sound right?

Reply:
The wording can obviously change, but what I'm getting at is that the comment reads like any one of the 3 items in the list is handled by the code below the comment, which is not true

Reply:
I'm trying to update the comments now but fighting with npm-check.

Reply:
god speed

Reply:
Updated!  Let me know how that looks.